_studio/shared/mfx_trace/src/mfx_trace_utils_linux.cpp

```diff
@@ -21,26 +21,58 @@
 
 #include "mfx_trace_utils.h"
 #include <sstream>
-#include <iomanip>
+#include <cstring>
 
 #ifdef MFX_TRACE_ENABLE
 #include "vm_sys_info.h"
 
-#if defined(ANDROID)
-#include "snprintf_s.h"
-#endif
+inline std::string& trim_string(std::string&& s)
+{
+    // Remove leading spaces, tabulation, = character
+    size_t pos = s.find_first_not_of(" =\t");
+    if (pos != std::string::npos)
+        s.erase(0, pos);
+
+    // Remove trailing spaces, tabulation, newline symbol, spaces
+    pos = s.find_last_not_of(" \t\n\r\f\v");
+    if (pos != std::string::npos)
+        s.erase(pos + 1);
```

Comment:
should not that be `s.erase(pos+1, s.end())`? And what will happen if `pos+1=s.end()`.

Reply:
no, I'm calling
```
erase(size_t index = 0, size_t count = npos)
```
so pos is not an iterator but index. In my case it will just remove all characters in string started from first of any of " \t\n\r\f\v" occurrence  (in our simple par-file format it is ok).

This function works fine if pos == str.size() it just do nothing. it throws out_of_range exception if pos > str.size() but it shouldn't happen in this code

Reply:
I am ok then.